util.c

```diff
@@ -1932,6 +1932,50 @@ int mdmon_running(const char *devnm)
 	return 0;
 }
 
+/*
+ * wait_for_mdmon_control_socket() - Waits for mdmon control socket
+ * to be created within specified time.
+ * @container_devnm: Device for which mdmon control socket should start.
+ *
+ * In foreground mode, when mdadm is trying to connect to control
+ * socket it is possible that the mdmon has not created it yet.
+ * Give some time to mdmon to create socket. Timeout set to 2 sec.
+ *
+ * Return: MDADM_STATUS_SUCCESS if connect succeed, otherwise return
+ * error code.
+ */
+mdadm_status_t wait_for_mdmon_control_socket(const char *container_devnm)
+{
+	enum mdadm_status status = MDADM_STATUS_SUCCESS;
+	int sfd, rv, retry_count = 0;
+	struct sockaddr_un addr;
+	char path[PATH_MAX];
+
+	snprintf(path, PATH_MAX, "%s/%s.sock", MDMON_DIR, container_devnm);
+	sfd = socket(PF_LOCAL, SOCK_STREAM, 0);
+	if (!is_fd_valid(sfd))
+		return MDADM_STATUS_ERROR;
+
+	addr.sun_family = PF_LOCAL;
+	strcpy(addr.sun_path, path);
```

Comment:
Static analysis reported:
`string_overflow You might overrun the 108-character destination string addr.sun_path by writing 4096 characters from path`
please change it to `strncpy`, eventually `snprintf` (to ensure null ending`.